cognite/neat/workflows/steps/step_model.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar, Type, TypeVar
+from pydantic import BaseModel, ConfigDict
+from cognite.neat.stores.metrics import NeatMetricsCollector
+
+from cognite.neat.workflows.model import WorkflowConfigItem
+
+
+class Config(BaseModel):
+    ...
+
+
+class DataContract(BaseModel):
+    model_config: ClassVar[ConfigDict] = ConfigDict(arbitrary_types_allowed=True)
+
+
+T_Input = TypeVar("T_Input", bound=DataContract)
+
+T_Output = TypeVar("T_Output", bound=DataContract)
+
+
+class Step(ABC):
+    description: str = ""
+    category: str = "default"
+    configuration_templates: list[WorkflowConfigItem] = []
+    scope: str = "global"
+
+    def __init__(self, metrics: NeatMetricsCollector, data_store_path: str | None = None):
+        self.log: bool = False
+        self.data_store_path: str = data_store_path
+        # Comment: Metrics should be optional
+        self.metrics: NeatMetricsCollector = metrics or None
+
+    def set_flow_context(self, context: dict[str, Type[DataContract]]):
+        self.flow_context = context
```

Comment:
We have discussed this before, lets not have a `__init__` in the base `Step` this leads to increased complexity for `Step` authors.